### PR TITLE
Update modules to pass the correct variable to `ModuleIO`

### LINKED ISSUE #294: Bug with fault tolerance check if no models are generated
```
[2022-01-28 10:14:58,661 libutil ERROR] float division by zero
Traceback (most recent call last):
  File "/Users/rodrigo/repos/haddock3/src/haddock/libs/libutil.py", line 320, in log_error_and_exit
    yield
  File "/Users/rodrigo/repos/haddock3/src/haddock/clis/cli.py", line 148, in main
    workflow.run()
  File "/Users/rodrigo/repos/haddock3/src/haddock/libs/libworkflow.py", line 27, in run
    step.execute()
  File "/Users/rodrigo/repos/haddock3/src/haddock/libs/libworkflow.py", line 100, in execute
    module.run(**self.config)
  File "/Users/rodrigo/repos/haddock3/src/haddock/modules/base_cns_module.py", line 48, in run
    self._run()
  File "/Users/rodrigo/repos/haddock3/src/haddock/modules/refinement/mdref/__init__.py", line 101, in _run
    faulty = io.check_faulty()
  File "/Users/rodrigo/repos/haddock3/src/haddock/libs/libontology.py", line 168, in check_faulty
    faulty_per = (1 - (present / total)) * 100
ZeroDivisionError: float division by zero
```

### PR DESCRIPTION
**Pull request by rvhonorato**
This PR closes #294 

~I did not run any sort of tests, not sure how to run them since it changed in #291.~
Result of the test below

### CODE REVIEW

**Comment by rvhonorato** [comment]
Figured out how to run the test, I get this output:

```text
[2022-01-28 10:41:42,968 cli INFO] 
##############################################
#                                            #
#                 HADDOCK 3                  #
#                                            #
##############################################

Starting HADDOCK 3.0.0 on 2022-01-28 10:41:00

Python 3.9.0 | packaged by conda-forge | (default, Nov 26 2020, 07:54:06) 
[Clang 11.0.0 ]

[2022-01-28 10:41:44,919 libworkflow INFO] Reading instructions of [topoaa] step
[2022-01-28 10:41:44,919 libworkflow INFO] Reading instructions of [rigidbody] step
[2022-01-28 10:41:44,919 libworkflow INFO] Reading instructions of [caprieval] step
[2022-01-28 10:41:44,919 libworkflow INFO] Reading instructions of [seletop] step
[2022-01-28 10:41:44,919 libworkflow INFO] Reading instructions of [flexref] step
[2022-01-28 10:41:44,920 libworkflow INFO] Reading instructions of [caprieval.1] step
[2022-01-28 10:41:44,920 libworkflow INFO] Reading instructions of [emref] step
[2022-01-28 10:41:44,920 libworkflow INFO] Reading instructions of [caprieval.2] step
[2022-01-28 10:41:44,963 base_cns_module INFO] Running [topoaa] module
[2022-01-28 10:41:44,965 __init__ INFO] [topoaa] Molecule 1: cro.pdb
[2022-01-28 10:41:44,966 __init__ INFO] [topoaa] Sanitizing molecule cro.pdb
[2022-01-28 10:41:44,971 __init__ INFO] [topoaa] Topology CNS input created in cro.inp
[2022-01-28 10:41:44,971 __init__ INFO] [topoaa] Molecule 2: dna.pdb
[2022-01-28 10:41:44,972 __init__ INFO] [topoaa] Sanitizing molecule dna.pdb
[2022-01-28 10:41:44,978 __init__ INFO] [topoaa] Topology CNS input created in dna.inp
[2022-01-28 10:41:44,978 __init__ INFO] [topoaa] Running CNS Jobs n=2
[2022-01-28 10:41:44,978 libparallel INFO] Using 2 cores
[2022-01-28 10:41:46,337 libparallel INFO] >> /cro.inp completed 50% 
[2022-01-28 10:41:46,994 libparallel INFO] >> /dna.inp completed 100% 
[2022-01-28 10:41:46,995 libparallel INFO] 2 tasks finished
[2022-01-28 10:41:46,995 __init__ INFO] [topoaa] CNS jobs have finished
[2022-01-28 10:41:46,998 base_cns_module INFO] Module [topoaa] finished.
[2022-01-28 10:41:47,029 base_cns_module INFO] Running [rigidbody] module
[2022-01-28 10:41:47,031 __init__ INFO] [rigidbody] crossdock=true
[2022-01-28 10:41:47,031 __init__ INFO] [rigidbody] Preparing jobs...
[2022-01-28 10:41:47,094 __init__ INFO] [rigidbody] Running CNS Jobs n=20
[2022-01-28 10:41:47,094 libparallel INFO] Using 3 cores
[2022-01-28 10:42:18,287 libparallel INFO] >> /rigidbody_1.inp completed 5% 
[2022-01-28 10:42:18,301 libparallel INFO] >> /rigidbody_2.inp completed 10% 
[2022-01-28 10:42:18,301 libparallel INFO] >> /rigidbody_3.inp completed 15% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_4.inp completed 20% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_5.inp completed 25% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_6.inp completed 30% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_7.inp completed 35% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_8.inp completed 40% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_9.inp completed 45% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_10.inp completed 50% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_11.inp completed 55% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_12.inp completed 60% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_13.inp completed 65% 
[2022-01-28 10:42:18,302 libparallel INFO] >> /rigidbody_14.inp completed 70% 
[2022-01-28 10:42:18,303 libparallel INFO] >> /rigidbody_15.inp completed 75% 
[2022-01-28 10:42:18,303 libparallel INFO] >> /rigidbody_16.inp completed 80% 
[2022-01-28 10:42:18,303 libparallel INFO] >> /rigidbody_17.inp completed 85% 
[2022-01-28 10:42:18,303 libparallel INFO] >> /rigidbody_18.inp completed 90% 
[2022-01-28 10:42:18,303 libparallel INFO] >> /rigidbody_19.inp completed 95% 
[2022-01-28 10:42:18,303 libparallel INFO] >> /rigidbody_20.inp completed 100% 
[2022-01-28 10:42:18,303 libparallel INFO] 20 tasks finished
[2022-01-28 10:42:18,303 __init__ INFO] [rigidbody] CNS jobs have finished
[2022-01-28 10:42:18,363 base_cns_module INFO] Module [rigidbody] finished.
[2022-01-28 10:42:18,367 __init__ INFO] Running [caprieval] module
[2022-01-28 10:42:18,404 capri INFO] Using sequence alignment
[2022-01-28 10:42:18,409 capri INFO] Sequence identity of chain A is 100.00%
[2022-01-28 10:42:18,411 capri INFO] Sequence identity of chain B is 100.00%
[2022-01-28 10:42:19,265 __init__ INFO] [caprieval] Calculating FNAT
[2022-01-28 10:42:19,265 __init__ INFO] [caprieval]  cutoff: 5.0A
[2022-01-28 10:42:19,375 __init__ INFO] [caprieval] Calculating I-RMSD
[2022-01-28 10:42:19,375 __init__ INFO] [caprieval]  cutoff: 10.0A
[2022-01-28 10:42:19,523 __init__ INFO] [caprieval] Calculating L-RMSD
[2022-01-28 10:42:19,659 __init__ INFO] [caprieval] Calculating I-L-RMSD
[2022-01-28 10:42:19,660 __init__ INFO] [caprieval]  cutoff: 10.0A
[2022-01-28 10:42:19,987 __init__ INFO] [caprieval] Calculating DockQ metric
[2022-01-28 10:42:19,987 __init__ INFO] [caprieval]  Saving output to capri.tsv
[2022-01-28 10:42:19,992 __init__ INFO] Module [caprieval] finished.
[2022-01-28 10:42:19,996 __init__ INFO] Running [seletop] module
[2022-01-28 10:42:19,999 __init__ INFO] Module [seletop] finished.
[2022-01-28 10:42:20,032 base_cns_module INFO] Running [flexref] module
[2022-01-28 10:42:20,055 __init__ INFO] [flexref] Running CNS Jobs n=5
[2022-01-28 10:42:20,056 libparallel INFO] Using 3 cores
[2022-01-28 10:42:52,923 libparallel INFO] >> /flexref_1.inp completed 20% 
[2022-01-28 10:42:52,924 libparallel INFO] >> /flexref_2.inp completed 40% 
[2022-01-28 10:43:17,383 libparallel INFO] >> /flexref_3.inp completed 60% 
[2022-01-28 10:43:17,383 libparallel INFO] >> /flexref_4.inp completed 80% 
[2022-01-28 10:43:17,383 libparallel INFO] >> /flexref_5.inp completed 100% 
[2022-01-28 10:43:17,383 libparallel INFO] 5 tasks finished
[2022-01-28 10:43:17,383 __init__ INFO] [flexref] CNS jobs have finished
[2022-01-28 10:43:17,387 libutil ERROR] 20.00% of output was not generated for this module and tolerance was set to 5.00%.
Traceback (most recent call last):
  File "/Users/rodrigo/repos/haddock3/src/haddock/libs/libutil.py", line 320, in log_error_and_exit
    yield
  File "/Users/rodrigo/repos/haddock3/src/haddock/clis/cli.py", line 148, in main
    workflow.run()
  File "/Users/rodrigo/repos/haddock3/src/haddock/libs/libworkflow.py", line 27, in run
    step.execute()
  File "/Users/rodrigo/repos/haddock3/src/haddock/libs/libworkflow.py", line 100, in execute
    module.run(**self.config)
  File "/Users/rodrigo/repos/haddock3/src/haddock/modules/base_cns_module.py", line 48, in run
    self._run()
  File "/Users/rodrigo/repos/haddock3/src/haddock/modules/refinement/flexref/__init__.py", line 106, in _run
    self.finish_with_error(_msg)
  File "/Users/rodrigo/repos/haddock3/src/haddock/modules/__init__.py", line 156, in finish_with_error
    raise RuntimeError(reason)
RuntimeError: 20.00% of output was not generated for this module and tolerance was set to 5.00%.
[2022-01-28 10:43:17,391 libutil ERROR] 20.00% of output was not generated for this module and tolerance was set to 5.00%.
[2022-01-28 10:43:17,391 libutil ERROR] An error has occurred, see log file. And contact the developers if needed.
[2022-01-28 10:43:17,392 libutil INFO] Finished at 28/01/2022 10:43:17. For any help contact us at https://github.com/haddocking/haddock3/issues. Dovidenia! Tot ziens! Ciao!.

************************* DOCKING-PROTEIN-DNA-TEST.CFG *************************
```

The information provided is correct, 4 models are generated out of 5 expected, 1 of them fail with this error:

```
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H5T "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -O5' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C5' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H5' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H5''"
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C4' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H4' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -O4' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C1' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H1' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -N1  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C6  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H6  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C2  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -O2  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -N3  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H3  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C4  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -O4  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C5  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C7  "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H71 "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H72 "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H73 "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C2' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H2' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H2''"
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H3' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -C3' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -O3' "
 %ATMCHK-ERR: unknown coordinates for atom "B   -35  -DT  -H3T "
 %ATMCHK error encountered: Unknown coordinates
   (CNS is in mode: SET ABORT=NORMal END)
 *****************************************************
 ABORT mode will terminate program execution. 
 *****************************************************
 Program will stop immediately.
          ============================================================
           Maximum dynamic memory allocation:  2989673080 bytes
           Maximum dynamic memory overhead:          5344 bytes
           Program started at: 10:47:54 on 28-Jan-2022
           Program stopped at: 10:47:57 on 28-Jan-2022
           CPU time used:       1.0390 seconds
          ============================================================
```

So the workflow-integration test is failing but the implementation is correct, how to proceed?

**Comment by amjjbonvin** [comment]
For testing purposes set the tolerance to 20%

**Comment by rvhonorato** [comment]
> For testing purposes set the tolerance to 20%

Where is best to set this? in the `.cfg` or in the `run_test.py`?

**Comment by amjjbonvin** [comment]
In the `-test.cfg` file I would say - just to make it work

But I don’t have those failures in the protein-DNA example

File: src/haddock/libs/libontology.py [hunk 1]
```diff
@@ -165,6 +165,12 @@ def check_faulty(self):
                 if element.is_present():
                     present += 1
 
+        if total == 0:
+            _msg = ("No outputs were expected to exist, "
```
Comment (amjjbonvin) [inline src/haddock/libs/libontology.py:169]: Not sure what this message intends to say.... No outputs were expected to exists - it sounds as if you don't expect any output

Comment (rvhonorato) [reply src/haddock/libs/libontology.py:169]: that is right, if this gets triggered means you did not pass anything

Comment (amjjbonvin) [reply src/haddock/libs/libontology.py:169]: Then the message is misleading. Should rather read something like:
`None of the expected output was passed`

Comment (rvhonorato) [reply src/haddock/libs/libontology.py]: Changed!

**Review by amjjbonvin — APPROVED** [review]

**Review by joaomcteixeira — APPROVED** [review]
:+1:

**Pull request merged**